ext/vscode/src/commands/env.ts

```diff
@@ -9,42 +9,134 @@ import { quickPickWorkspaceFolder } from '../utils/quickPickWorkspaceFolder';
 import { EnvironmentInfo, getAzDevTerminalTitle, getEnvironments } from './cmdUtil';
 import { executeInTerminal } from '../utils/executeInTerminal';
 import { spawnAsync } from '../utils/process';
+import { isTreeViewModel, TreeViewModel } from '../utils/isTreeViewModel';
+import { AzureDevCliEnvironments } from '../views/workspace/AzureDevCliEnvironments';
+import { AzureDevCliEnvironment } from '../views/workspace/AzureDevCliEnvironment';
 
-export async function selectEnvironment(context: IActionContext, selectedFile?: vscode.Uri): Promise<void> {
+export async function editEnvironment(context: IActionContext, selectedEnvironment?: TreeViewModel): Promise<void> {
+    if (selectedEnvironment) {
+        const environment = selectedEnvironment.unwrap<AzureDevCliEnvironment>();
+
+        if (environment.environmentFile) {
+            const document = await vscode.workspace.openTextDocument(environment.environmentFile);
+
+            await vscode.window.showTextDocument(document);
+        }
+    }
+}
+
+export async function deleteEnvironment(context: IActionContext, selectedItem?: vscode.Uri | TreeViewModel): Promise<void> {
+    const selectedEnvironment = isTreeViewModel(selectedItem) ? selectedItem.unwrap<AzureDevCliEnvironment>() : undefined;
+    const selectedFile = isTreeViewModel(selectedItem) ? selectedItem.unwrap<AzureDevCliEnvironments>().context.configurationFile : selectedItem;
     let folder: vscode.WorkspaceFolder | undefined = (selectedFile ? vscode.workspace.getWorkspaceFolder(selectedFile) : undefined);
     if (!folder) {
         folder = await quickPickWorkspaceFolder(context, localize('azure-dev.commands.util.needWorkspaceFolder', "To run '{0}' command you must first open a folder or workspace in VS Code", 'env select'));
     }
     const cwd = folder.uri.fsPath;
 
-    let envData: EnvironmentInfo[] = [];
-    let errorMsg: string | undefined = undefined;
-    try {
-        envData = await getEnvironments(context, cwd);
-    } catch(err) {
-        errorMsg = parseError(err).message;
-        // Treated the same as no environments case
+    let name = selectedEnvironment?.name;
+
+    if (!name) {       
+        let envData: EnvironmentInfo[] = [];
+        try {
+            envData = await getEnvironments(context, cwd);
+        } catch(err) {
+            // Treated the same as no environments case
```

Comment:
Below (`selectEnvironment` action) we do display the error from `getEnvironments()` call. Is there a reason why we suppress it here? Should the error be available as part of diagnostics output?

Reply:
I'm not sure, this is code that @philliphoff wrote that I haven't touched. Phil, do you remember why it's done this way?

Reply:
My guess is that I just took the original `selectEnvironment()` command as a template for the `deleteEnvironment()` command.  That error handling hasn't changed since its original commit.

Reply:
I made it output the error message to the log output channel in both cases. For the select case, since it shows a prompt with details, I also show the error there (that code isn't changed), but for the info message shown in delete, there isn't a "detail" notion, so the output message will be the only place for that error.